SPECS-EXTENDED/liblouis/liblouis.spec

```diff
@@ -0,0 +1,407 @@
+honor of Louis Braille. It features support for computer and literary braille,
+supports contracted and uncontracted translation for many languages and has
+support for hyphenation. New languages can easily be added through tables that
+support a rule- or dictionary based approach. Liblouis also supports math
+braille (Nemeth and Marburg).
+
+Liblouis has features to support screen-reading programs. This has led to its
+use in two open-source screen readers, NVDA and Orca. It is also used in some
+commercial assistive technology applications for example by ViewPlus.
+
+Liblouis is based on the translation routines in the BRLTTY screen reader for
+Linux. It has, however, gone far beyond these routines.
+
+%package        devel
+Summary:        Development files for %{name}
+License:        LGPLv3+
+Requires:       %{name}%{?_isa} = %{version}-%{release}
+Requires:       pkgconfig
+
+%description    devel
+The %{name}-devel package contains libraries and header files for
+developing applications that use %{name}.
+
+%package        utils
+Summary:        Command-line utilities to test %{name}
+License:        GPLv3+
+Requires:       %{name}%{?_isa} = %{version}-%{release}
+
+%description    utils
+Six test programs are provided as part of the liblouis package. They
+are intended for testing liblouis and for debugging tables. None of
+them is suitable for braille transcription.
+
+%package -n python3-louis
+%{?python_provide:%python_provide python3-louis}
+Summary:        Python 3 language bindings for %{name}
+License:        LGPLv3+
+Requires:       %{name} = %{version}-%{release}
+Obsoletes:      %{name}-python3 < 2.6.2-3
+Provides:       %{name}-python3 = %{version}-%{release}
+BuildArch:      noarch
+
+%description -n python3-louis
+This package provides Python 3 language bindings for %{name}.
+
+%if %{with docs}
+%package doc
+Summary:        Documentation for %{name}
+License:        LGPLv3+
+Requires:       %{name} = %{version}-%{release}
+BuildArch:      noarch
+
+%description doc
+This package provides the documentation for liblouis.
+%endif
+
+%global __os_install_post %{nil}
+%global debug_package %{nil}
+License:        LGPLv3+
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+
+%prep
+%autosetup
+
+
+%build
+%configure --disable-static --enable-ucs4
+# parallel builds fail
+make
+%if %{with docs}
+cd doc; xetex %{name}.texi
+%endif
+
+
+%check
+make check
+
+
+%install
+%make_install
+rm -f %{buildroot}/%{_infodir}/dir
+rm -f %{buildroot}/%{_libdir}/%{name}.la
+rm -rf %{buildroot}/%{_bindir}/lou_maketable*
+rm -rf %{buildroot}/%{_defaultdocdir}/%{name}/
+
+# Replace table files with identical content by symlinks
+%fdupes -s %{buildroot}%{_datadir}/%{name}/tables/
+
+cd python/louis
+
+install -d %{buildroot}%{python3_sitelib}/louis
+install -pm 0644 __init__.py %{buildroot}%{python3_sitelib}/louis/
+%py_byte_compile %{__python3} %{buildroot}%{python3_sitelib}/louis/
```

Comment:
I'm not sure I understand what's happening here- why are we disabling automatic compilation as the beginning of the spec, and then doing it manually here?

Reply:
I have analyzed further on this. It seems the following guideline is used by the people from where we imported our spec file initially for the various reasons.

"One MUST disable their automatic compilation, and compile them explicitly by the %py_byte_compile macro if and only if the bytecompilation is desired."

But when it comes to Mariner I believe we no need to follow the above guideline and hence I removed the following code. 
 `%py_byte_compile %{__python3} %{buildroot}%{python3_sitelib}/louis/`

After successful build, I have observed that In our case the respective *.py files are automatically byte compiled which is what we wanted.  
`rpm -qlp RPMS/noarch/python3-louis-3.24.0-2.cm2.noarch.rpm
/usr/lib/python3.9/site-packages/louis
/usr/lib/python3.9/site-packages/louis/__init__.py
/usr/lib/python3.9/site-packages/louis/__pycache__
/usr/lib/python3.9/site-packages/louis/__pycache__/__init__.cpython-39.opt-1.pyc
/usr/lib/python3.9/site-packages/louis/__pycache__/__init__.cpython-39.pyc
`